pkg/k8s/watchers/pod.go

```diff
@@ -263,10 +270,19 @@ func (k *K8sWatcher) updateK8sPodV1(oldK8sPod, newK8sPod *slim_corev1.Pod) error
 		return k.deleteK8sPodV1(newK8sPod)
 	}
 
-	// The pod IP can never change, it can only switch from unassigned to
-	// assigned
-	// Process IP updates
-	k.addK8sPodV1(newK8sPod)
+	if newK8sPod.Spec.HostNetwork {
+		logger.Debug("Pod is using host networking")
+		return nil
+	}
+
+	oldPodIPs := k8sUtils.ValidIPs(oldK8sPod.Status)
+	newPodIPs := k8sUtils.ValidIPs(newK8sPod.Status)
+	err := k.updatePodHostData(oldK8sPod, newK8sPod, oldPodIPs, newPodIPs)
```

Comment:
I don't understand how this is fixing the issue you've tagged since `updatePodHostData` doesn't do any diff of old and new pod data. If for example only pod labels had changed, then with this change we would still end up calling `updatePodHostData` unnecessarily?

Reply:
In `updateK8sPodV1` we don't call `k.redirectPolicyManager.OnAddPod(pod)`, also now we check if the pod spec is the same or not in lines 603-608.

Reply:
>also now we check if the pod spec is the same or not in lines 603-608.

My comment still stands - 

>If for example only pod labels had changed, then with this change we would still end up calling updatePodHostData unnecessarily?

If only pod labels have changed, and not pod IPs, then we shouldn't be calling `updatePodHostData`. Label/annotation updates are handled in `updateK8sPodV1` separately. Please check my comment - https://github.com/cilium/cilium/issues/13136#issuecomment-694357756.